### PR TITLE
tools/bluestore_tool: add support for zap-device command

### PR DESCRIPTION
**Pull request by ifed01**
As BlueStore might have multiple bdev labels on a single device it might be trickier to zap OSD device(s) using dd tool. So let's implement this in ceph-bluestore-tool and update ceph-volume accordingly. The latter is out of scope of this PR though.

Please note that this PR relies on 
https://github.com/ceph/ceph/pull/59627
to fix --yes-i-really-really-mean-it option specification .

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by ifed01** [comment]
jenkins test make check

**Comment by guits** [comment]
jenkins test make check

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -8988,6 +8988,55 @@ void BlueStore::trim_free_space(const string& type, std::ostream& outss)
...
+int BlueStore::zap_device(CephContext* cct, const string& dev, uint64_t gap_size)
+{
+  string path; // dummy var for dout
+  dout(5)<<  __func__ << " " << dev << dendl;
+  auto * _bdev =
+    BlockDevice::create(cct, dev, nullptr, nullptr, nullptr, nullptr);
+  int r = _bdev->open(dev);
+  if (r < 0)
+    goto fail;
+  if (!gap_size) {
```
Comment (aclamk) [inline src/os/bluestore/BlueStore.cc:9000]: I think any freedom around selecting zap brush size (gap_size) is redundant.
We want to brush just and exactly as much as needed, which is:
`max(BDEV_LABEL_BLOCK_SIZE, bdev->get_block_size())`

My proposal is to fix it.

Comment (guits) [reply src/os/bluestore/BlueStore.cc:9000]: For what it's worth, this is only my opinion, but what I expect from the `zap-device` command is simply to remove the Bluestore signature and nothing else.
I wouldn't really care about the block size being written, as long as the Bluestore signatures are removed altogether. Your call.

**Comment by ifed01** [comment]
Closed in favor of #59762

**Pull request closed without merging**